addons/account/models/account_move_line.py

```diff
@@ -556,15 +556,16 @@ def _compute_account_id(self):
                 elif line.move_id.is_purchase_document(include_receipts=True):
                     line.account_id = accounts['expense'] or line.account_id
             elif line.partner_id:
-                line.account_id = self.env['account.account']._order_by_frequency_per_partner(
-                    company_id=line.company_id.id,
-                    partner_id=line.partner_id.id,
-                    move_type=line.move_id.move_type,
-                    limit=1,
-                )[0]
+                if self.env['account.move'].search_count([('partner_id', '=', line.partner_id.id), ('company_id', '=', line.company_id.id)]):
```

Comment:
Shouldn't you put `HAVING COUNT() > 0` instead to avoid a query on the big table?